spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/Issue2120Test.java

```diff
@@ -1,29 +1,21 @@
 package edu.umd.cs.findbugs.detect;
 
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnJre;
+import org.junit.jupiter.api.condition.JRE;
 
 import edu.umd.cs.findbugs.AbstractIntegrationTest;
 import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcher;
 import edu.umd.cs.findbugs.test.matcher.BugInstanceMatcherBuilder;
 
 import static edu.umd.cs.findbugs.test.CountMatcher.containsExactly;
-import static org.junit.Assume.assumeFalse;
-import static org.junit.Assume.assumeThat;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.is;
-import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 
-public class Issue2120Test extends AbstractIntegrationTest {
-    @Before
-    public void verifyJavaVersion() {
-        assumeFalse(System.getProperty("java.specification.version").startsWith("1."));
-        int javaVersion = Integer.parseInt(System.getProperty("java.specification.version"));
-        assumeThat(javaVersion, is(greaterThanOrEqualTo(11)));
-    }
+class Issue2120Test extends AbstractIntegrationTest {
 
     @Test
-    public void test() {
+    @DisabledOnJre(JRE.JAVA_11)
```

Comment:
See the files used under test, they only work on jdk 17 of the build.  So yes its meant to skip.  The important thing is that the build as it is was using toolchains and not adhering to what it had stated.  While some of these were jdk 14 added or what not, it only matters what we build against as we store those in singular packages to not get out of control.  So anything noted with 14 is simply able to build at 17.  

You would not see the specific issue on this PR, it shows when jdk 11 is then used to build since all the files used for test were flagged only working with higher jdk.

Reply:
The test was assuming `assumeThat(javaVersion, is(greaterThanOrEqualTo(11)));` so it ran on JDK 11 but now its not: `@DisabledOnJre({ JRE.JAVA_8, JRE.JAVA_11 })`

Reply:
Yes to both here.  Due to the folder being used, I added the flag in that way to reflect that.  Will review all that though and squash the tests back into normal folders and use junit flags for same as necessary.  That will clean up the build a bit.  Not sure we should prolong this PR though to do that work.  At the moment, these are all flagged to generally match the folders we claim the tests to be in per the jdk we use to build.  Possibly this is a follow up PR that addresses properly each introduced jdk usage directly and just cleans up the gradle build scripts for same.  Otherwise I fear this continues to grow and makes it harder to review than it already is.

Let me know what makes most sense there.

Reply:
Looks good to me, thank you!